src/span.h

```diff
@@ -73,11 +105,8 @@ class Span
  *
  * std::span will have a constructor that implements this functionality directly.
  */
-template<typename A, int N>
-constexpr Span<A> MakeSpan(A (&a)[N]) { return Span<A>(a, N); }
-
 template<typename V>
-constexpr Span<typename std::remove_pointer<decltype(std::declval<V>().data())>::type> MakeSpan(V& v) { return Span<typename std::remove_pointer<decltype(std::declval<V>().data())>::type>(v.data(), v.size()); }
+constexpr auto MakeSpan(V&& v) -> Span<typename std::remove_pointer<decltype(v.data())>::type> { return Span<typename std::remove_pointer<decltype(v.data())>::type>(v.data(), v.size()); }
```

Comment:
@sipa @jb55 I think passing any rvalue/temporary here will result in dangling pointers.

I did check the docs and the [c++ 20 span](https://en.cppreference.com/w/cpp/container/span/span) has a ctor that takes `R&&` ...

```c++
template <class R>
constexpr span(R&& r);
```
however It says R must model [borrowed_range](https://en.cppreference.com/w/cpp/ranges/borrowed_range). 
In other words R must be a non owning view (like basic_string_view, span, ref_view ...).
```
The behavior is undefined if R does not actually model contiguous_range and
sized_range or if R does not model borrowed_range while element_type is non-const.
```






```suggestion
constexpr auto MakeSpan(V& v) -> Span<typename std::remove_pointer<decltype(v.data())>::type> { return Span<typename std::remove_pointer<decltype(v.data())>::type>(v.data(), v.size()); }
```

Reply:
I've made some changes, and commented below.